### PR TITLE
Allow to initialize the ``find-mode`` task from previously obtained importance samples

### LINKED ISSUE #633: Allow to initialize the ``find-mode`` task from previously obtained importance samples
Useful in general, needed for @cbolognani's project.

### PR DESCRIPTION
**Pull request by dvandyk**
Resolves #633.

### CODE REVIEW

File: python/eos/analysis.py [hunk 1]
```diff
@@ -419,8 +419,8 @@ def sample_pmc(self, log_proposal, step_N=1000, steps=10, final_N=5000, rng=np.r
             The parameter determines the number of update steps to "look back".
             The default value of 1 disables this feature, a value of 0 means that all previous steps are used.
 
-        :return: A tuple of the parameters as array of length N = step_N * steps + final_N, the (linear) weights as array of length N, and the
-            final proposal function as pypmc.density.mixture.MixtureDensity.
+        :return: A tuple of the parameters as array of length N = step_N * steps + final_N, the (linear) weights as array of length N, the posterior values as array of length N, and the
+            final proposal function as pypmc.density.mixture.MixtureDensity.s
```
Comment (mreboud) [inline python/eos/analysis.py:423]: This ``s`` is weird

File: python/eos/data/native.py [hunk 2]
```diff
@@ -426,11 +431,18 @@ def create(path, parameters, samples, weights):
...
 
+        if not posterior_values is None and not samples.shape[0] == posterior_values.shape[0]:
+            raise RuntimeError(f'Shape of posterior values {posterior_values.shape} incompatible with shape of samples {samples.shape}')
+
         os.makedirs(path, exist_ok=True)
         with open(os.path.join(path, 'description.yaml'), 'w') as description_file:
             yaml.dump(description, description_file, default_flow_style=False)
         _np.save(os.path.join(path, 'samples.npy'), samples)
         _np.save(os.path.join(path, 'weights.npy'), weights)
+        eos.error('before writing posterior values')
```
Comment (mreboud) [inline python/eos/data/native.py:442]: This error message is weird

File: python/eos/data/native.py [hunk 3]
```diff
@@ -426,11 +431,18 @@ def create(path, parameters, samples, weights):
...
+        if not posterior_values is None:
+            eos.error('writing posterior values')
```
Comment (mreboud) [inline python/eos/data/native.py:444]: ditto

**Review by mreboud — CHANGES_REQUESTED** [review]
There are a few typos :)

File: python/eos/analysis.py (hunk 1 above)
Comment (dvandyk) [reply python/eos/analysis.py:423]: Good catch. My stupid home-office keyboard at work again.

File: python/eos/data/native.py (hunk 3 above)
Comment (dvandyk) [reply python/eos/data/native.py:444]: Fixed.

**Review by mreboud — APPROVED** [review]

**Pull request merged**